test/cache.ts

```diff
@@ -310,3 +310,31 @@ test('does not hang on huge response', withServer, async (t, server, got) => {
 
 	t.is(body.length, bufferSize * times);
 });
+
+test('cached response ETag', withServer, async (t, server, got) => {
+	const etag = 'foobar';
+	const body = 'responseBody';
+
+	server.get('/', (request, response) => {
+		if (request.headers['if-none-match'] === etag) {
+			response.writeHead(304, {ETag: 'asdf'});
+			response.end();
+		} else {
+			response.writeHead(200, {ETag: etag});
+			response.end(body);
+		}
+	});
+
+	const cache = new Map();
+
+	const originalResponse = await got({cache});
+
+	t.false(originalResponse.isFromCache);
+	t.is(originalResponse.body, body);
+	t.is(cache.size, 1);
```

Comment:
This test fails with a value of `0`, not `1`.

Reply:
I know, I wrote it in the commit message.
I think it's some kind of bug of got and not of the test.

Reply:
When the first request `await got({cache})` is resolved, the cache is not immediately populated with the response.

Reply:
From `cacheable-request` documentation:
> .on('request', request)
> request event to get the request object of the request.
>
> This event will only fire if an HTTP request is actually made, not when a response is retrieved from cache. However, you should always handle the request event to end the request and handle any potential request errors.

"you should always handle the request event to end the request"
So i ended the request as suggested and it worked. Also the the promise is now resolved with the actual response and not the request.